DDOC: Add documentation to src/mars.d

Geod24 opened the pull request:
Figured it would be the best / easiest place to start :)

Geod24 committed:
DDOC: Add documentation to src/mars.d

Geod24 commented:
Ping ? This is trivial (just DDOC improvement), but it's still wasting the auto-tester ressources.

WalterBright commented on src/mars.d:
```diff
@@ -1,10 +1,15 @@
-// http://www.digitalmars.com
-// Distributed under the Boost Software License, Version 1.0.
-// http://www.boost.org/LICENSE_1_0.txt
+/**
```
Please add the line:
```
 * Compiler implementation of the $(LINK2 http://www.dlang.org, D programming language)
```

Geod24 replied:
IMO it's a bit superfluous in the docs, as someone coming accross this code will most likely do it on Github or the dlang website. Even if the person didn't knew the origin of the file, it would be trivial to google it.
Nevertheless, I added it, and we can revisit if there is a need when more modules documentation gets added.

Geod24 commented:
Rebased on master, updated:

```diff
--- a/src/mars.d
+++ b/src/mars.d
@@ -1,5 +1,7 @@
 /**
- * Entry point for DMD
+ * Entry point for DMD.
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language)
  *
  * This modules defines the entry point (main) for DMD, as well as related
```

Geod24 commented:
Ping ?

ibuclaw commented on src/mars.d:
```diff
@@ -66,6 +82,14 @@ extern (C++) const(char)* toWinPath(const(char)* src)
+ *   loc = The line number information from where the call originates
+ *   f = a `ddmd.root.file.File` handle to read
+ */
 extern (C++) void readFile(Loc loc, File* f)
```
These functions need a new home before documenting them?

Geod24 replied:
They should definitely get a new home, but I don't see how that should block them from being documented.

WalterBright commented:
Auto-merge toggled on
